std/file.d

```diff
@@ -581,9 +601,18 @@ version(Windows) private void writeImpl(const(char)[] name, const(FSChar)* namez
         cenforce(h != INVALID_HANDLE_VALUE, name, namez);
         scope(exit) cenforce(CloseHandle(h), name, namez);
         DWORD numwritten;
-        cenforce(WriteFile(h, buffer.ptr, to!DWORD(buffer.length), &numwritten, null) != 0
-                && buffer.length == numwritten,
-                name, namez);
+        immutable size = buffer.length;
+        size_t cnt, sum;
+        size_t buff;
+        while (sum != size)
+        {
+            cnt = (size - sum < 4096) ? (size - sum) : 4096;
+            WriteFile(h, buffer.ptr + sum, cnt, &numwritten, null);
+            if (numwritten != cnt)
+                break;
+            sum += numwritten;
+        }
+        cenforce(sum == size, name, namez);
```

Comment:
I see three copies of the same logic. Maybe instead of hand-writing the code for each platform, use `std.stdio.File` (`data.chunks(...).put(File(...).lockingBinaryWriter)`) OSLT.